service/history/workflow/retry.go

```diff
@@ -53,7 +53,7 @@ import (
 
 // TODO treat 0 as 0, not infinite
 
-func getBackoffInterval(
+var getBackoffInterval = func(
```

Comment:
I get why this is done but I really don't like making this a mutable global. I'd want a more experienced server dev to look at this but it makes me uncomfortable

Reply:
+1 to Tim's comment. Maybe we could refactor this logic into its own object? e.g. convert `BackoffIntervalCalculator` to an interface with a `GetBackoffInterval` method. Then we could provide a different implementation for tests. WDYT?

Reply:
Oh, I knew you would not like it. First about refactoring, if you want to refactor you need to be sure that code you change is covered by tests. Working with code for the past week I am not confident at all. Second, this is unexported function, plus it has type. You can't accidentally change it also you can set it to nil. But the same true for any place where function is used as a value. We have tests and code reviews to catch such situations.

Reply:
@pdoerner [down with getters that start with `Get`](https://go.dev/doc/effective_go#Getters)

Reply:
I wish proto compiler follow this. Btw, maybe we should tweak it?

Reply:
Changing `func` to be a method is not a refactoring. I would call it "code reorg". You need to extract interface. Either this PR or another is ok.

Reply:
> I wish proto compiler follow this. Btw, maybe we should tweak it?

We can’t: the variables are already publicly exported so we can’t also export their name as a function :(.

Reply:
Oh.. that's why they used `Get` prefix.